### PR TITLE
added query filter status

### LINKED ISSUE #118: Check filters status
@alnitak, How do you now track filter statuses between hot restarts?
i.e. I'm following codelab from google io 2024, and there were a line that we could check status with `isFilterActive(filter)` method, which is now deprecated.

_Originally posted by @misterfourtytwo in https://github.com/alnitak/flutter_soloud/issues/108#issuecomment-2309004690_

### PR DESCRIPTION
**Pull request by alnitak**
## Description

- Addded query filter status to fix #118 ie:
```
bool isEchoActive = SoLoud.instance.filters.echoFilter.isActive();
```

- Chore: reset callback function pointers on dispose on C side. Probably unnecessary and irrelevant but it could solve #117

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [x] 🗑️ Chore

### CODE REVIEW

**Comment by filiph** [comment]
I have a single comment (about naming of `isActive()`) and two nits.

**Comment by alnitak** [comment]
> I have a single comment (about naming of `isActive()`) and two nits.

Thanks to your valuable suggestion, I added the abstract class `FilterBase` common to all filter classes.
Also, the method docs are now visible!

I also forgot to add a `timeStretch()` method in the `PitchShiftSingle` filter. It just calculates the value for the `shift` based on the requested speed (it acts like 1.5x, 2x etc from YouTube without changing the timbre of the voice).

File: lib/src/filters/bassboost_filter.dart [hunk 1]
```diff
@@ -23,19 +23,12 @@ enum BassBoostEnum {
       };
 }
 
-abstract class BassBoostInternal {
-  const BassBoostInternal(SoundHash? soundHash) : _soundHash = soundHash;
+abstract class BassBoostInternal extends FilterBase {
```
Comment (filiph) [inline lib/src/filters/bassboost_filter.dart:26]: [nit] If these `SomeFilterInternal` classes really are supposed to be internal, then I'd suggest:

1. Making the classes private (by renaming to `_BassBoostInternal`), OR
2. If they cannot be made private (i.e. only accessible within the same file), then maybe mark them as `@internal` or `@visibleForTesting`

If they're more like Base classes (publicly visible), then they could be renamed to something like `BassBoostBase`.

**Review by filiph — APPROVED** [review]
This is great! LGTM.

I added a suggestion but if you want, you can do that later or never.

File: lib/src/filters/bassboost_filter.dart (hunk 1 above)
Comment (alnitak) [reply lib/src/filters/bassboost_filter.dart:26]: done.

**Pull request merged**